### ISSUE TITLE
How to determine the parameters used for correspondence searching?

### ISSUE THREAD

**Issue by shawnshuailin** [IntelVCL/Open3D#371]
In the program for colored point cloud registration, the parameters used for voxel downsampling and correspondence searching should be relevant to point cloud resolution, how to determine these parameters? Thanks!

**Comment by qianyizh** [comment]
TL;DR:  This is an empirical problem. We usually suggest correspondence search radius to be roughly 1x~2x voxel size used to down sample the target point cloud.

Long version:
This is an empirical problem. Sometimes the input point clouds are uneven. Sometimes the point clouds are too large. We first use ``voxel_down_sample()`` to make it (roughly) uniformly sampled at a controllable scale (a few thousands to tens of thousands points).

Upon registration, consider a perfect alignment, the correspondence search radius needs to be large enough so that the points in overlapping region can always find a correspondence. On the other hand, it should not be too large; otherwise the points in the non-overlapping region will find (incorrect) correspondences that create distraction.

A rule to determine the correspondence search radius I often use is to align the point cloud with itself: the search radius should barely allow each point to find its nearest neighbor in the point cloud. For example:
```
from open3d import *
x = read_point_cloud("fragment.ply")
y = voxel_down_sample(x, 0.03)
dis = sorted(compute_point_cloud_nearest_neighbor_distance(y))
print(dis[int(len(dis) * 0.95)])
# ~0.03
```
The 95% longest distance between neighbors is roughly the same as voxel size.

So when the target point clouds is voxel down sampled, use roughly 1x~2x voxel size as the search radius.

This is actually a very good question. @syncle is there anything you want to add? This may be worth to add into FAQ.

**Comment by syncle** [comment]
This is very good. Thanks @qianyizh! I think in this way, we can automate parameter setting from the point cloud.

**Comment by shawnshuailin** [comment]
Thanks very much for your answer, @qianyizh ! I will have a try on my own data.
